front-end-tools/src/Main.tsx

```diff
@@ -0,0 +1,579 @@
+                                            type="button"
+                                            onClick={() => {
+                                                setTxHashDeploy('');
+                                                setTransactionErrorDeploy('');
+                                                const tx = deploy(connection, account, base64Module);
+                                                tx.then(setTxHashDeploy).catch((err: Error) =>
+                                                    setTransactionErrorDeploy((err as Error).message)
+                                                );
+                                            }}
+                                        >
+                                            Deploy smart contract module
+                                        </button>
+                                        <br />
+                                        <br />
+                                    </>
+                                )}
+                                {!txHashDeploy && transactionErrorDeploy && (
+                                    <div className="alert alert-danger" role="alert">
+                                        Error: {transactionErrorDeploy}.
+                                    </div>
+                                )}
+                                {txHashDeploy && (
+                                    <>
+                                        Transaction hash (May take a moment to finalize): {}
+                                        <a
+                                            className="link"
+                                            target="_blank"
+                                            rel="noreferrer"
+                                            href={`https://${
+                                                isTestnet ? `testnet.` : ``
+                                            }ccdscan.io/?dcount=1&dentity=transaction&dhash=${txHashDeploy}`}
+                                        >
+                                            {txHashDeploy}
+                                        </a>
+                                    </>
+                                )}
+                            </TestBox>
+                            <TestBox header="Step 2: Initialize Smart Contract">
+                                <label className="field">
+                                    Add Module Reference:
+                                    <br />
+                                    <input
+                                        className="inputFieldStyle"
+                                        id="moduleReference"
+                                        type="text"
+                                        placeholder="91225f9538ac2903466cc4ab07b6eb607a2cd349549f357dfdf4e6042dde0693"
+                                        onChange={changeModuleReferenceHandler}
+                                    />
```

Comment:
Users can now copy the module reference from the front end.

Reply:
The front end has now a `checkBox` that you can click `Use Module Reference from Step 1`.

It will then autofill the `moduleReference` form Step 1 to Step2. In detail, it will first use `moduleReferenceDeployed` (meaning the `moduleReference` that you just deployed to chain in Step 1 (or if that value is not available) the `moduleReferenceCalculated` if you just uploaded a module file in Step 1 but did not deploy it to chain.